bmi_sdk/bmi_sdk/bmi_variable.py

```diff
@@ -0,0 +1,128 @@
+from __future__ import annotations
+
+from enum import Enum
+
+# Once python 3.8 is no longer supported, just get Annotated directly
+# from typing import Annotated, Any, Literal, Optional, Self, Union
+from typing import Any, Literal, Optional, Union
+
+import numpy as np
+from pydantic import (
+    BaseModel,
+    ConfigDict,
+    Field,
+    RootModel,
+    computed_field,
+    model_validator,
+)
+
+# Only needed on python 3.8
+from typing_extensions import Annotated, Self
+
+from .bmi_grid import Grid
+
+Integer = RootModel[Annotated[str, Literal["int", "int16", "int32", "integer"]]]
+Long = RootModel[Annotated[str, Literal["int", "int64", "longlong"]]]
+Float = RootModel[
+    Annotated[
+        str,
+        Literal[
+            "float",
+            "float32",
+            "np.float32",
+            "numpy.float32",
+            "np.single",
+            "numpy.single",
+        ],
+    ]
+]
+Double = RootModel[
+    Annotated[str, Literal["float", "float64", "np.float64", "numpy.float64"]]
+]
+
+VarType = Union[Integer, Long, Float, Double]
+
+
+class VarLocation(str, Enum):
```

Comment:
So this would be `Location`

Reply:
Changed to GridLocation in 290c33a5cac997eb10ea1808810243abcd844939